src/flatcollections-array/FlatArray/FlatArray.T/Functional/Functional.Chunk.cs

```diff
@@ -0,0 +1,40 @@
+namespace System;
+
+partial struct FlatArray<T>
+{
+    public FlatArray<FlatArray<T>> Chunk(int size)
+    {
+        if (size < 1)
+        {
+            throw InnerExceptionFactory.SizeOutsideBounds(nameof(size), size);
+        }
+
+        if (length == default)
+        {
+            return default;
+        }
+
+        var chunksLength = length / size;

```

Comment:
Would like suggest avoiding declaration of intermediate variable `chunksLength`, in a way like following.

(To avoid possible ambiguity between `chunksLength` and `chunkLength`, and to emphasize `chunks.Length` is the effective length.)

```cs
var chunks = length % size == default
    ? new FlatArray<T>[length / size]
    : new FlatArray<T>[length / size + 1];
```

Reply:
Fixed